File: linked-list/1669.js

var mergeInBetween = function (list1, a, b, list2) {
    let left = null
    let right = list1
    for (let i = 0; i <= b; i++) {
        if (i === a - 1) {
            left = right
        }
        right = right.next
    }
    left.next = list2
    let temp = list2
    while (temp.next) {
        temp = temp.next
    }
    temp.next = right
    return list1
};

File: linked-list/1669.py

class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        left = None
        right = list1
        for i in range(b+1):
            if i == a - 1:
                left = right
            right = right.next
        left.next = list2
        temp = list2
        while temp.next:
            temp = temp.next
        temp.next = right
        return list1